From 7b3256ea07fff1740c07c81ccbf3d864e8de2a5d Mon Sep 17 00:00:00 2001
From: nluo <ning.luo@intel.com>
Date: Tue, 13 Aug 2013 15:24:43 +0800
Subject: Bugfix for video editor output artifacts caused by improper frame drops.

In the original comparison,
1. pC->pC1->VideoAU.m_CTS is got from metadata of mediabuffer as below
mMediaBuffer->meta_data()->findInt64(kKeyTime,(int64_t*)&tempTime64) /1000
since the kKeyTime is of the us unit so we need do a us to ms conversion, this will lead to a bottom egde align.
2. iCts is got through ((M4OSA_Int32)pC->ewc.dInputVidCts) - pC->pC1->iVoffset - 1;
This needs a double to int conversion.
So in some cases, this compare need at most 1ms adjustment.

Category: AOSP Enhancement
Domain: MEDIA
Upstream-Candidate: yes

Change-Id: I4eb0e01423b8f98fd0a39bde42f53a0f4334f84e
Signed-off-by: nluo <ning.luo@intel.com>
---
 libvideoeditor/vss/src/M4VSS3GPP_EditVideo.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/libvideoeditor/vss/src/M4VSS3GPP_EditVideo.c b/libvideoeditor/vss/src/M4VSS3GPP_EditVideo.c
index f19f412..bfdcc97 100755
--- a/libvideoeditor/vss/src/M4VSS3GPP_EditVideo.c
+++ b/libvideoeditor/vss/src/M4VSS3GPP_EditVideo.c
@@ -266,7 +266,7 @@ M4OSA_ERR M4VSS3GPP_intEditStepVideo( M4VSS3GPP_InternalEditContext *pC )
                 /**
                 * If the AU is good to be written, write it, else just skip it */
                 if( ( M4OSA_FALSE == bSkipFrame)
-                    && (( pC->pC1->VideoAU.m_CTS >= iCts)
+                    && (( (pC->pC1->VideoAU.m_CTS + 1) >= iCts)
                     && (pC->pC1->VideoAU.m_CTS < iNextCts)
                     && (pC->pC1->VideoAU.m_size > 0)) )
                 {
-- 
1.7.4.1

